fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/info/MTMVPartitionDefinition.java

```diff
@@ -0,0 +1,225 @@
+            Expr expr = convertToLegacyAutoPartitionExprs(Lists.newArrayList(functionCallExpression))
+                    .get(0);
+            partitionColName = getColNameFromExpr(expr);
+            mtmvPartitionInfo.setExpr(expr);
+        } else {
+            partitionColName = this.partitionCol;
+        }
+        mtmvPartitionInfo.setPartitionCol(partitionColName);
+        RelatedTableInfo relatedTableInfo = getRelatedTableInfo(planner, ctx, logicalQuery, partitionColName);
+        mtmvPartitionInfo.setRelatedCol(relatedTableInfo.getColumn());
+        mtmvPartitionInfo.setRelatedTable(relatedTableInfo.getTableInfo());
+        if (this.partitionType == MTMVPartitionType.EXPR) {
+            try {
+                MTMVPartitionExprFactory.getExprSerice(mtmvPartitionInfo.getExpr()).analyze(mtmvPartitionInfo);
+            } catch (org.apache.doris.common.AnalysisException e) {
+                throw new AnalysisException(e.getMessage(), e);
+            }
+        }
+        return mtmvPartitionInfo;
+    }
+
+    /**
+     * getColNameFromExpr
+     *
+     * @param expr expr
+     * @return String
+     */
+    public static String getColNameFromExpr(Expr expr) {
+        if (!(expr instanceof FunctionCallExpr)) {
+            throw new AnalysisException(
+                    "auto create partition only support function call expr is: "
+                            + MTMVPartitionInfo.MTMV_PARTITION_FUNCTIONS);
+        }
+        FunctionCallExpr functionCallExpr = (FunctionCallExpr) expr;
+        List<Expr> paramsExpr = functionCallExpr.getParams().exprs();
+        String name = functionCallExpr.getFnName().getFunction();
+        if (MTMVPartitionInfo.MTMV_PARTITION_FUNCTIONS.contains(name)) {
+            for (Expr param : paramsExpr) {
+                if (param instanceof SlotRef) {
+                    return ((SlotRef) param).getColumnName();
+                }
+            }
+            throw new AnalysisException("can not find colName");
+        } else {
+            throw new AnalysisException(
+                    "auto create partition only support function call expr is: "
+                            + MTMVPartitionInfo.MTMV_PARTITION_FUNCTIONS);
+        }
+    }
+
+    private RelatedTableInfo getRelatedTableInfo(NereidsPlanner planner, ConnectContext ctx, LogicalPlan
+            logicalQuery,
+            String partitionColName) {
+        CascadesContext cascadesContext = planner.getCascadesContext();
+        SessionVariable sessionVariable = cascadesContext.getConnectContext().getSessionVariable();
+        Set<String> tempDisableRules = sessionVariable.getDisableNereidsRuleNames();
+        // Should not make table without data to empty relation when analyze the related table,
+        // so add disable rules
+        sessionVariable.setDisableNereidsRules(CreateMTMVInfo.MTMV_PLANER_DISABLE_RULES);
+        cascadesContext.getStatementContext().invalidCache(SessionVariable.DISABLE_NEREIDS_RULES);
```

Comment:
@seawinde more than one place use disable rule. This approach has poor maintainability, we need to think of a better solution.